Branch-SDK/Branch-SDK/BranchShareLink.m

```diff
@@ -0,0 +1,253 @@
+            getLongURLWithParams:self.serverParameters
+            andChannel:self.linkProperties.channel
+            andTags:self.linkProperties.tags
+            andFeature:self.linkProperties.feature
+            andStage:self.linkProperties.stage
+            andAlias:self.linkProperties.alias];
+    NSURL *URL = [[NSURL alloc] initWithString:URLString];
+    item = [[BranchShareActivityItem alloc] initWithPlaceholderItem:URL];
+    item.itemType = BranchShareActivityItemTypeBranchURL;
+    item.parent = self;
+    [items addObject:item];
+
+    if (self.shareObject) {
+        item = [[BranchShareActivityItem alloc] initWithPlaceholderItem:self.shareObject];
+        item.itemType = BranchShareActivityItemTypeOther;
+        item.parent = self;
+        [items addObject:self.shareObject];
+    }
+
+    _activityItems = items;
+    return _activityItems;
+}
+
+- (void) presentActivityViewControllerFromViewController:(UIViewController*_Nullable)viewController
+                                                  anchor:(UIBarButtonItem*_Nullable)anchor {
+
+    UIActivityViewController *shareViewController =
+        [[UIActivityViewController alloc]
+            initWithActivityItems:self.activityItems
+            applicationActivities:nil];
+    shareViewController.title = self.title;
+
+    if ([shareViewController respondsToSelector:@selector(completionWithItemsHandler)]) {
+        shareViewController.completionWithItemsHandler =
+        ^ (NSString *activityType, BOOL completed, NSArray *returnedItems, NSError *activityError) {
+            [self shareDidComplete:completed activityError:activityError];
+        };
+    } else {
+        #pragma clang diagnostic push
+        #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+        shareViewController.completionHandler =
+        ^ (UIActivityType activityType, BOOL completed) {
+            [self shareDidComplete:completed activityError:nil];
+        };
+        #pragma clang diagnostic pop
+    }
+
+    if (self.linkProperties.controlParams[BRANCH_LINK_DATA_KEY_EMAIL_SUBJECT]) {
+        @try {
+            [shareViewController
+                setValue:self.linkProperties.controlParams[BRANCH_LINK_DATA_KEY_EMAIL_SUBJECT]
+                forKey:@"subject"];
+        }
+        @catch (NSException *exception) {
+            NSLog(@"Warning: Unable to setValue 'emailSubject' forKey 'subject' on UIActivityViewController.");
+        }
+    }
+
+    UIViewController *presentingViewController = nil;
+    if ([viewController respondsToSelector:@selector(presentViewController:animated:completion:)]) {
+        presentingViewController = viewController;
+    } else {
+        Class UIApplicationClass = NSClassFromString(@"UIApplication");
+        UIViewController *rootController = [UIApplicationClass sharedApplication].delegate.window.rootViewController;
+        if ([rootController respondsToSelector:@selector(presentViewController:animated:completion:)]) {
```

Comment:
Since the `viewController` argument is nullable, it seems like this could carelessly be called from some random place in the app while another VC is modally presented from the rootViewController. To address that kind of issue, I did this in the RN code: https://github.com/BranchMetrics/react-native-branch-deep-linking/blob/master/ios/RNBranch.m#L92, walking the chain of presentedViewControllers until the end is found. I did that in a couple of apps before, and it worked well.